### PR TITLE
fix(Table): solved the issue with rowspan and striped 

### PR DESCRIPTION
**Pull request by emreekiziltoprak**
## Description

When using rowspan in a table and also applying stripe, the rows corresponding to the rowspan should be striped with the same color as the rowspan. For example, if the leftmost column occupies 3 rows while the other columns only occupy one row, those 3 rows should remain the same color when using stripe.

## Type of Change

- [ ] 🆕 New Component
- [ ] ✨ Feature / Enhancement
- [ x] 🐛 Bug Fix
- [ ] 💥 Breaking Change
- [ ] 📦 Build / Dependency / Docs Update
- [ ] 🧹 Code Refactoring

## Checklist

- [x ] Self-reviewed, no leftover logs or debug code
- [ ] Uses design tokens — no hardcoded style values
- [ ] Accessible (keyboard nav, ARIA, contrast)
- [x ] Tests added/updated and passing
- [x ] Storybook story added/updated

## How to Test

```
"use client";

import Table from "@/components/Table";

type MaintenanceRow = {
  span: number;
  equipment: string;
  location: string;
  servicedAt: string;
  finding: string;
  action: string;
  owner: string;
  dueDate: string;
  status: string;
};

const group = (
  span: number,
  equipment: string,
  location: string,
  servicedAt: string,
  findings: [string, string, string, string, string][],
): MaintenanceRow[] =>
  findings.map(([finding, action, owner, dueDate, status], index) => ({
    span: index === 0 ? span : 1,
    equipment,
    location,
    servicedAt,
    finding,
    action,
    owner,
    dueDate,
    status,
  }));

const data: MaintenanceRow[] = [
  ...group(3, "EQ-1042", "Hat 1 / Presleme", "12.03.2026", [
    ["Hidrolik hatta basınç düşüşü gözlendi", "Sızdırmazlık contası değiştirildi", "A. Yılmaz", "20.03.2026", "Tamamlandı"],
    ["Yağ sıcaklığı eşik değerin üzerinde", "Soğutma fanı temizliği yapıldı", "A. Yılmaz", "22.03.2026", "Tamamlandı"],
    ["Titreşim sensörü kalibrasyon dışı", "Sensör yeniden kalibre edilecek", "M. Demir", "05.04.2026", "Devam ediyor"],
  ]),
  ...group(2, "EQ-1043", "Hat 1 / Kesim", "14.03.2026", [
    ["Bıçak aşınması tolerans sınırında", "Bıçak seti yenilendi", "S. Kaya", "18.03.2026", "Tamamlandı"],
    ["Koruma kapağı kilidi geç tepki veriyor", "Emniyet rölesi değişimi planlandı", "S. Kaya", "10.04.2026", "Beklemede"],
  ]),
  ...group(3, "EQ-1051", "Hat 2 / Kaynak", "19.03.2026", [
    ["Kaynak akımında dalgalanma", "Güç ünitesi kartı değiştirildi", "E. Aydın", "25.03.2026", "Tamamlandı"],
    ["Gaz debisi düşük okunuyor", "Regülatör revizyona gönderildi", "E. Aydın", "02.04.2026", "Devam ediyor"],
    ["Torç kablosunda yalıtım aşınması", "Kablo demeti komple yenilenecek", "B. Şahin", "15.04.2026", "Beklemede"],
  ]),
  ...group(2, "EQ-1077", "Hat 2 / Montaj", "21.03.2026", [
    ["Konveyör hız sapması %4", "Sürücü parametreleri güncellendi", "B. Şahin", "24.03.2026", "Tamamlandı"],
    ["Rulman sesi normalin üzerinde", "Rulman değişimi için parça bekleniyor", "M. Demir", "12.04.2026", "Beklemede"],
  ]),
  ...group(3, "EQ-1080", "Hat 3 / Boyama", "26.03.2026", [
    ["Filtre doluluk oranı %92", "Filtre kaseti değiştirildi", "Z. Öztürk", "28.03.2026", "Tamamlandı"],
    ["Nem oranı proses aralığının dışında", "Nem alma ünitesi devreye alındı", "Z. Öztürk", "01.04.2026", "Tamamlandı"],
    ["Püskürtme başlığında tıkanma", "Başlık sökülüp ultrasonik temizlik yapılacak", "A. Yılmaz", "18.04.2026", "Devam ediyor"],
  ]),
  ...group(2, "EQ-1095", "Hat 3 / Paketleme", "30.03.2026", [
    ["Etiket okuyucu hatalı okuma veriyor", "Optik kafa temizlendi ve test edildi", "S. Kaya", "03.04.2026", "Tamamlandı"],
    ["Bantlama ünitesinde kaçık hizalama", "Mekanik ayar ve merkezleme yapılacak", "E. Aydın", "20.04.2026", "Beklemede"],
  ]),
];

const groupedColumn = (title: string, dataKey: string) => ({
  title,
  dataKey,
  rowSpan: (row: object) => (row as MaintenanceRow).span,
});

const columns = [
  groupedColumn("Ekipman", "equipment"),
  groupedColumn("Lokasyon", "location"),
  groupedColumn("Bakım Tarihi", "servicedAt"),
  { title: "Bulgu", dataKey: "finding" },
  { title: "Aksiyon", dataKey: "action" },
  { title: "Sorumlu", dataKey: "owner" },
  { title: "Termin", dataKey: "dueDate" },
  { title: "Durum", dataKey: "status" },
];

const Home = () => {
  return (
    <div style={{ margin: "0 auto", padding: 20 }}>
      <Table data={data} columns={columns} border="cellBorders" striped showFixedRowNumbers hoverable />
    </div>
  );
};

export default Home;

```

{B6501BAF-07E6-4AEF-99F3-CC0F00B59210}

### CODE REVIEW

**Review by atakanbayrak — CHANGES_REQUESTED** [review]
Image

I got the error with given page.tsx. Main issue pops up from TableBody useState import.

File: src/lib/components/Table/Table.tsx [hunk 1]
```diff
@@ -1,3 +1,5 @@
+"use client";
```
Comment (atakanbayrak) [inline src/lib/components/Table/Table.tsx]: I am not sure should be enable client rendering for Table component. I guess we talked this issue before and I remember that we decided to leave it server side component. Am I wrong ? @aktasmehmet

**Comment by ZehranurC** [comment]
> ## Description
> ## Type of Change
> * [ ]  🆕 New Component
> * [ ]  ✨ Feature / Enhancement
> * [ x] 🐛 Bug Fix
> * [ ]  💥 Breaking Change
> * [ ]  📦 Build / Dependency / Docs Update
> * [ ]  🧹 Code Refactoring
> 
> ## Screenshots / Preview
> ## Checklist
> * [x ] Self-reviewed, no leftover logs or debug code
> * [ ]  Uses design tokens — no hardcoded style values
> * [ ]  Accessible (keyboard nav, ARIA, contrast)
> * [x ] Tests added/updated and passing
> * [x ] Storybook story added/updated
> 
> ## How to Test
> ```
> "use client";
> 
> import Table from "@/components/Table";
> 
> type MaintenanceRow = {
>   span: number;
>   equipment: string;
>   location: string;
>   servicedAt: string;
>   finding: string;
>   action: string;
>   owner: string;
>   dueDate: string;
>   status: string;
> };
> 
> const group = (
>   span: number,
>   equipment: string,
>   location: string,
>   servicedAt: string,
>   findings: [string, string, string, string, string][],
> ): MaintenanceRow[] =>
>   findings.map(([finding, action, owner, dueDate, status], index) => ({
>     span: index === 0 ? span : 1,
>     equipment,
>     location,
>     servicedAt,
>     finding,
>     action,
>     owner,
>     dueDate,
>     status,
>   }));
> 
> const data: MaintenanceRow[] = [
>   ...group(3, "EQ-1042", "Hat 1 / Presleme", "12.03.2026", [
>     ["Hidrolik hatta basınç düşüşü gözlendi", "Sızdırmazlık contası değiştirildi", "A. Yılmaz", "20.03.2026", "Tamamlandı"],
>     ["Yağ sıcaklığı eşik değerin üzerinde", "Soğutma fanı temizliği yapıldı", "A. Yılmaz", "22.03.2026", "Tamamlandı"],
>     ["Titreşim sensörü kalibrasyon dışı", "Sensör yeniden kalibre edilecek", "M. Demir", "05.04.2026", "Devam ediyor"],
>   ]),
>   ...group(2, "EQ-1043", "Hat 1 / Kesim", "14.03.2026", [
>     ["Bıçak aşınması tolerans sınırında", "Bıçak seti yenilendi", "S. Kaya", "18.03.2026", "Tamamlandı"],
>     ["Koruma kapağı kilidi geç tepki veriyor", "Emniyet rölesi değişimi planlandı", "S. Kaya", "10.04.2026", "Beklemede"],
>   ]),
>   ...group(3, "EQ-1051", "Hat 2 / Kaynak", "19.03.2026", [
>     ["Kaynak akımında dalgalanma", "Güç ünitesi kartı değiştirildi", "E. Aydın", "25.03.2026", "Tamamlandı"],
>     ["Gaz debisi düşük okunuyor", "Regülatör revizyona gönderildi", "E. Aydın", "02.04.2026", "Devam ediyor"],
>     ["Torç kablosunda yalıtım aşınması", "Kablo demeti komple yenilenecek", "B. Şahin", "15.04.2026", "Beklemede"],
>   ]),
>   ...group(2, "EQ-1077", "Hat 2 / Montaj", "21.03.2026", [
>     ["Konveyör hız sapması %4", "Sürücü parametreleri güncellendi", "B. Şahin", "24.03.2026", "Tamamlandı"],
>     ["Rulman sesi normalin üzerinde", "Rulman değişimi için parça bekleniyor", "M. Demir", "12.04.2026", "Beklemede"],
>   ]),
>   ...group(3, "EQ-1080", "Hat 3 / Boyama", "26.03.2026", [
>     ["Filtre doluluk oranı %92", "Filtre kaseti değiştirildi", "Z. Öztürk", "28.03.2026", "Tamamlandı"],
>     ["Nem oranı proses aralığının dışında", "Nem alma ünitesi devreye alındı", "Z. Öztürk", "01.04.2026", "Tamamlandı"],
>     ["Püskürtme başlığında tıkanma", "Başlık sökülüp ultrasonik temizlik yapılacak", "A. Yılmaz", "18.04.2026", "Devam ediyor"],
>   ]),
>   ...group(2, "EQ-1095", "Hat 3 / Paketleme", "30.03.2026", [
>     ["Etiket okuyucu hatalı okuma veriyor", "Optik kafa temizlendi ve test edildi", "S. Kaya", "03.04.2026", "Tamamlandı"],
>     ["Bantlama ünitesinde kaçık hizalama", "Mekanik ayar ve merkezleme yapılacak", "E. Aydın", "20.04.2026", "Beklemede"],
>   ]),
> ];
> 
> const groupedColumn = (title: string, dataKey: string) => ({
>   title,
>   dataKey,
>   rowSpan: (row: object) => (row as MaintenanceRow).span,
> });
> 
> const columns = [
>   groupedColumn("Ekipman", "equipment"),
>   groupedColumn("Lokasyon", "location"),
>   groupedColumn("Bakım Tarihi", "servicedAt"),
>   { title: "Bulgu", dataKey: "finding" },
>   { title: "Aksiyon", dataKey: "action" },
>   { title: "Sorumlu", dataKey: "owner" },
>   { title: "Termin", dataKey: "dueDate" },
>   { title: "Durum", dataKey: "status" },
> ];
> 
> const Home = () => {
>   return (
>     <div style={{ margin: "0 auto", padding: 20 }}>
>       <Table data={data} columns={columns} border="cellBorders" striped showFixedRowNumbers hoverable />
>     </div>
>   );
> };
> 
> export default Home;
> ```
> 
> {B6501BAF-07E6-4AEF-99F3-CC0F00B59210}

**Comment by ZehranurC** [comment]
When editing PR content, I recommend that you don’t leave the description fields blank; when we review them before the review, it will give us a general idea of what you’ve done.

File: src/lib/components/Table/Table.tsx (hunk 1 above)
Comment (aktasmehmet) [reply src/lib/components/Table/Table.tsx]: Yes, I dont think this is necessary now.

File: src/app/layout.tsx [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import "./globals.css";
 import "../lib/styles/themes/default-theme.css";
 import type { ReactNode } from "react";
-import { MotifProvider } from "../lib";
+import MotifProvider from "../lib/motif/context/MotifProvider";
```
Comment (aktasmehmet) [inline src/app/layout.tsx]: This increases the complexity of review process for big PRs like this.

File: src/lib/components/Table/components/Body/TableBody.tsx [hunk 3]
```diff
@@ -31,7 +41,17 @@ const TableBody = memo((props: Props) => {
...
         </tr>
       ) : (
-        visibleRows.map((row, index) => <DataRow key={row.motifIndex} rowNumberStatic={index + 1} row={row} rowIndex={index} />)
+        visibleRows.map((row, index) => (
+          <DataRow
+            key={row.motifIndex}
+            rowNumberStatic={index + 1}
+            row={row}
+            rowIndex={index}
+            isStripe={stripeGroups !== undefined && stripeGroups[index] % 2 === 1}
```
Comment (aktasmehmet) [inline src/lib/components/Table/components/Body/TableBody.tsx]: Probably this is gonna work even if there will be a NaN


```suggestion
            isStripe={stripeGroups?.[index] % 2 === 1}
```

File: src/lib/components/Table/helper.ts [hunk 4]
```diff
@@ -86,6 +86,36 @@ export const getSpannedCellsMap = (columns: Column[], rows?: RowDetail[]): Spann
...
+
+  const groups: number[] = [];
+  let group = 0;
+  for (let rowIndex = 0; rowIndex < rowCount; rowIndex++) {
+    if (rowIndex > 0 && !continuationRows.has(rowIndex)) group++;
+    groups[rowIndex] = group;
+  }
+
+  return groups;
+};
```
Comment (aktasmehmet) [inline src/lib/components/Table/helper.ts]: If I am not wrong, this produces a possible (rowCount x span) complexity. This would be a better version. I've tested but can you also try?

```
export const getRowStripeGroups = (spannedCellsMap: SpannedCellsMap, rowCount: number) => {
  const rowEnds = Array.from(spannedCellsMap.entries()).reduce((map, [key, span]) => {
    if (!span || span.rowSpan <= 1) return map;

    const rowIndex = Number(key.split("-")[0]);
    const endRow = rowIndex + span.rowSpan - 1;
    map.set(rowIndex, Math.max(map.get(rowIndex) ?? rowIndex, endRow));

    return map;
  }, new Map<number, number>());

  return Array.from({ length: rowCount }).reduce(
    (state: { currentGroup: number; currentGroupEnd: number; groups: number[] }, _, rowIndex) => {
      const startsNewGroup = rowIndex > state.currentGroupEnd;
      const currentGroup = startsNewGroup ? state.currentGroup + 1 : state.currentGroup;
      const currentGroupEnd = Math.max(
        startsNewGroup ? rowIndex : state.currentGroupEnd,
        rowEnds.get(rowIndex) ?? (startsNewGroup ? rowIndex : state.currentGroupEnd),
      );

      state.groups[rowIndex] = currentGroup;

      return {
        currentGroup,
        currentGroupEnd,
        groups: state.groups,
      };
    },
    {
      currentGroup: 0,
      currentGroupEnd: 0,
      groups: new Array<number>(rowCount),
    },
  ).groups;
};
```

File: src/lib/components/Table/components/Body/TableBody.tsx [hunk 5]
```diff
@@ -1,19 +1,29 @@
...
 
 const TableBody = memo((props: Props) => {
-  const { loading, emptyMessage, striped } = props;
-  const { visibleRows, numberOfVisibleColumns } = useContext(TableContext);
+  const { loading, emptyMessage, striped, hoverable } = props;
+  const { visibleRows, numberOfVisibleColumns, spannedCellsMap } = useContext(TableContext);
+  const [hoveredRowIndex, setHoveredRowIndex] = useState<number>();
+
+  const stripeGroups = useMemo(
+    () => (striped ? getRowStripeGroups(spannedCellsMap, visibleRows?.length ?? 0) : undefined),
```
Comment (aktasmehmet) [inline src/lib/components/Table/components/Body/TableBody.tsx]:
```suggestion
    () => striped && getRowStripeGroups(spannedCellsMap, visibleRows?.length ?? 0),
```

File: src/lib/components/Table/components/Body/DataRow.tsx [hunk 6]
```diff
@@ -10,16 +10,19 @@ type Props = {
...
-  const { rowNumberStatic, row, rowIndex } = props;
+  const { rowNumberStatic, row, rowIndex, isStripe, hoveredRowIndex, onHover } = props;
   const { columns, showFixedRowNumbers, selectable, selectHandler, rowColorCallback, spannedCellsMap } = useContext(TableContext);
 
-  const className = sanitizeModuleClasses(styles, row.isSelected && "selected", rowColorCallback?.(row.data));
+  const className = sanitizeModuleClasses(styles, isStripe && "stripedRow", row.isSelected && "selected", rowColorCallback?.(row.data));
 
   return (
-    <tr className={className}>
+    <tr className={className} onMouseEnter={onHover && (() => onHover(rowIndex))} onMouseLeave={onHover && (() => onHover(undefined))}>
```
Comment (aktasmehmet) [inline src/lib/components/Table/components/Body/DataRow.tsx]: for the hover, I am not a fan of doing it with events if we can on the css side. So, I've found a much cleaner solution that you will possibly get rid of lots of stuff. Here it is:

1 - in DataRow file, just put a const that checks if the row has a column with rowspan and put "groupStart" class name to the tr if it has.

```
const isGroupStart = columns.some((_, cIndex) => (spannedCellsMap.get(`${rowIndex}-${cIndex}`)?.rowSpan ?? 1) > 1);
```

2 - put this selector into the scss file:

```
tbody tr:hover td,
  tbody tr.groupStart:has(~ tr:hover:not(.groupStart)):not(
  :has(~ tr.groupStart ~ tr:hover)
) td[rowspan] {
    background: purple !important;
  }
```

**Review by aktasmehmet — CHANGES_REQUESTED** [review]

File: src/lib/components/Table/helper.ts (hunk 4 above)
Comment (emreekiziltoprak) [reply src/lib/components/Table/helper.ts]: {89819DB6-2C04-453A-B6CE-04C988F936BA}
real cost is elsewhere: Array.from(spannedCellsMap.entries()) copies the entire map into an array, and the second reduce allocates a new state object ({ currentGroup, currentGroupEnd, groups: state.groups }) on every single iteration. That allocation/GC overhead outweighs the theoretical extra term by a wide margin — a plain for loop with Set.add is much cheaper for V8 to optimize.

Comment (aktasmehmet) [reply src/lib/components/Table/helper.ts]: You are absolutely right, It is my overthinking there. Resolving this one. good work 💪

File: src/lib/components/Table/Table.module.scss [hunk 7]
```diff
@@ -125,9 +125,11 @@ $row-color-variants: "primary", "secondary", "light", "success", "danger", "warn
       }
     }
   }
-
-  &.hoverable tbody tr:hover {
-    background-color: var(--theme-color-surface-light-default);
+  &.hoverable tbody {
+    tr:hover td,
+    tr.groupStart:has(~ tr:hover:not(.groupStart)):not(:has(~ tr.groupStart ~ tr:hover)) td[rowspan] {
```
[the remark this answers is not in the document]
Comment (aktasmehmet) [reply src/lib/components/Table/Table.module.scss]: This is a right concern. Here is the screenshot of the bug:

Image

append this to the data
```
{
    span: 1,
    equipment: "STANDALONE-1",
    location: "Hat 3 / Sevkiyat",
    servicedAt: "02.04.2026",
    finding: "Bariyer fotosel hizası kaymış",
    action: "Fotosel yeniden hizalandı",
    owner: "M. Demir",
    dueDate: "06.04.2026",
    status: "Tamamlandı",
  },
  {
    span: 1,
    equipment: "STANDALONE-2",
    location: "Hat 3 / Sevkiyat",
    servicedAt: "03.04.2026",
    finding: "Forklift şarj istasyonu arızalı",
    action: "Şarj ünitesi servise gönderildi",
    owner: "B. Şahin",
    dueDate: "10.04.2026",
    status: "Devam ediyor",
  },
```

File: src/lib/components/Table/Table.stories.tsx [hunk 8]
```diff
@@ -612,3 +612,111 @@ export const Rowspan: Story = {
     return <Table data={data} columns={columns} border="cellBorders" />;
   },
 };
+
+export const RowspanStriped: Story = {
```
Comment (aktasmehmet) [inline src/lib/components/Table/Table.stories.tsx]: I dont think we need a story like this. We have rowspan story, user can use stripe prop to see the result. So, this is the merge of both "2 props", nothing more. So we should remove this.

File: src/lib/components/Table/Table.test.tsx [hunk 9]
```diff
@@ -751,6 +751,111 @@ describe("Table", () => {
     expect(secondRowCells[0]).toHaveTextContent("25");
   });
 
+  it("should keep every row of a rowSpan block in the same stripe group", () => {
```
Comment (aktasmehmet) [inline src/lib/components/Table/Table.test.tsx:763]: I'll check the tests later. Probably there is going to be another one after resolving the Claude's finding.

**Review by atakanbayrak — DISMISSED** [review]

File: src/lib/components/Table/Table.test.tsx [hunk 10]
```diff
@@ -751,6 +751,111 @@ describe("Table", () => {
...
+  it("should not mark a row as groupStart when no rowSpan is present", () => {
+    const { getTableBody } = renderExt(
+      <Table
+        columns={[
+          { title: "Name", dataKey: "name" },
+          { title: "Age", dataKey: "age" },
+        ]}
+        data={[
+          { name: "John", age: 30 },
+          { name: "Jane", age: 25 },
```
Comment (aktasmehmet) [inline src/lib/components/Table/Table.test.tsx]: These 2 tests here are actually technical steps to validate the usage. Tests are written to accomplish a functional result that user is affected. So, marking a row as groupStart or not has nothing to do with the user. User only see the hover correctly working or not.

So these 2 tests should be deleted. But I know this checks a bug that we are dealing with for a while. So, you can move this scenario into a single test like `should hover the grouped column along with the hovered row together when hoverable is true and there is a rowspan`.

File: src/lib/components/Table/Table.test.tsx [hunk 11]
```diff
@@ -751,6 +751,111 @@ describe("Table", () => {
...
+    );
+
+    const rows = Array.from(getTableBody().children) as HTMLTableRowElement[];
+    expect(rows[0]).not.toHaveClass("stripedRow");
+    expect(rows[1]).toHaveClass("stripedRow");
+    expect(rows[2]).not.toHaveClass("stripedRow");
+    expect(rows[3]).toHaveClass("stripedRow");
+  });
+
+  it("should not stripe rows when striped prop is omitted", () => {
```
Comment (aktasmehmet) [inline src/lib/components/Table/Table.test.tsx]: Dont need to test the `negative`version of a feature. There is already a test that checks the striped feature.

But, having said that, you should update that test. It is only checking striped class of the table but now we also put stripedRow class. That should also be checked in that test.

File: src/lib/components/Table/Table.test.tsx [hunk 12]
```diff
@@ -751,6 +751,111 @@ describe("Table", () => {
...
+        striped
+      />,
+    );
+
+    const rows = Array.from(getTableBody().children) as HTMLTableRowElement[];
+    rows.slice(0, 3).forEach(row => expect(row).not.toHaveClass("stripedRow"));
+    rows.slice(3, 6).forEach(row => expect(row).toHaveClass("stripedRow"));
+  });
+
+  it("should fall back to alternating every row when no rowSpan is present", () => {
```
Comment (aktasmehmet) [inline src/lib/components/Table/Table.test.tsx]: This test checks the negative version of the test above. It is not needed.

**Review by aktasmehmet — CHANGES_REQUESTED** [review]

File: src/lib/components/Table/helper.ts [hunk 13]
```diff
@@ -86,6 +86,34 @@ export const getSpannedCellsMap = (columns: Column[], rows?: RowDetail[]): Spann
...
+
+  const groups: number[] = [];
+  Array.from({ length: rowCount }).reduce((group: number, _, rowIndex) => {
+    const nextGroup = rowIndex > 0 && !continuationRows.has(rowIndex) ? group + 1 : group;
+    groups[rowIndex] = nextGroup;
+    return nextGroup;
+  }, 0);
+
+  return groups;
+};
```
Comment (aktasmehmet) [inline src/lib/components/Table/helper.ts]: What about merging them like this?:

```
export const getRowStripeGroups = (spannedCellsMap: SpannedCellsMap, rowCount: number) => {
  const continuation = Array(rowCount).fill(false);

  for (const [key, span] of spannedCellsMap) {
    if (!span || span.rowSpan <= 1) continue;

    const row = Number.parseInt(key, 10);
    for (let i = 1; i < span.rowSpan && row + i < rowCount; i++) {
      continuation[row + i] = true;
    }
  }

  return Array.from({ length: rowCount }).reduce<number[]>((groups, _, row) => {
    groups[row] = row === 0 ? 0 : groups[row - 1] + Number(!continuation[row]);
    return groups;
  }, []);
};
```

File: src/lib/components/Table/Table.stories.tsx [hunk 14]
```diff
@@ -612,3 +612,27 @@ export const Rowspan: Story = {
     return <Table data={data} columns={columns} border="cellBorders" />;
   },
 };
+
+export const RowspanOverlapping: Story = {
```
Comment (aktasmehmet) [inline src/lib/components/Table/Table.stories.tsx]: This story is not needed. User can see the result. There is nothing special to show. So we can remove this.

File: src/lib/components/Table/Table.test.tsx [hunk 15]
```diff
@@ -755,6 +760,30 @@ describe("Table", () => {
     expect(secondRowCells[0]).toHaveTextContent("25");
   });
```
Comment (aktasmehmet) [inline src/lib/components/Table/Table.test.tsx:761]: We need to add another test to check the rowspan cell is not hovered

**Review by aktasmehmet — CHANGES_REQUESTED** [review]

File: src/lib/components/Table/Table.test.tsx (hunk 15 above)
Comment (emreekiziltoprak) [reply src/lib/components/Table/Table.test.tsx:761]: The rowspan hover exclusion is pure css, so it is not possible to jest can assert here

**Review by aktasmehmet — APPROVED** [review]
🚀
